packages/bot/src/components/flyout.jsx

```diff
@@ -1,81 +1,105 @@
+            {
+                is_help_content ?
+                    <HelpBase nodes={block_nodes} /> :
+                    Object.keys(flyout_content).map((key, index) => {
+                        const nodes = flyout_content[key];
+                        const node = nodes[0];
+                        const tag_name = node.tagName.toUpperCase();
 
-                switch (tag_name) {
-                    case Blockly.Xml.NODE_BLOCK:
-                        return (
-                            <FlyoutBlock
-                                key={node.getAttribute('type') + index}
-                                id={`flyout__item-workspace--${index}`}
-                                block_node={node}
-                            />
-                        );
-                    case Blockly.Xml.NODE_LABEL:
-                        return (
-                            <div
-                                key={node.getAttribute('text') + index}
-                                className='flyout__item-label'
-                            >
-                                { node.getAttribute('text') }
-                            </div>
-                        );
-                    case Blockly.Xml.NODE_BUTTON: {
-                        const callback_key = node.getAttribute('callbackKey');
-                        const callback = Blockly.derivWorkspace.getButtonCallback(callback_key) || (() => {});
-                
-                        return (
-                            <button
-                                key={`${callback_key}${index}`}
-                                className='flyout__button'
-                                onClick={(button) => {
-                                    const flyout_button = button;
-        
-                                    // Workaround for not having a flyout workspace.
-                                    // eslint-disable-next-line no-underscore-dangle
-                                    flyout_button.targetWorkspace_ = Blockly.derivWorkspace;
-                                    // eslint-disable-next-line no-underscore-dangle
-                                    flyout_button.getTargetWorkspace = () => flyout_button.targetWorkspace_;
-        
-                                    callback(flyout_button);
-                                }}
-                            >
-                                { node.getAttribute('text') }
-                            </button>
-                        );
-                    }
-                    default:
-                        return null;
-                }
-            }) }
+                        switch (tag_name) {
+                            case Blockly.Xml.NODE_BLOCK: {
+                                const block_type = node.getAttribute('type');
+                                return (
+                                    <FlyoutBlock
+                                        key={`${block_type} ${Math.random()}`}
```

Comment:
The reason block_type + index is not usable is because of the issue "flyout not refreshed after new function added". Dynamic block_type like procedures_callnoreturn couldn't refresh if the component is cached. I will modified the code so that it only handle specifically for dynamic blocks.